src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -126,13 +132,15 @@ export class ElasticsearchService implements CoreService<ElasticsearchServiceSet
     return {
       legacy: { config$: clients$.pipe(map(clients => clients.config)) },
 
-      adminClient$: clients$.pipe(map(clients => clients.adminClient)),
-      dataClient$: clients$.pipe(map(clients => clients.dataClient)),
+      forPlugin: () => ({
```

Comment:
Why service should know who and how uses API?
what if other core service wants to use any `elasticsearch service` api? Say, we re-write `SavedObject` and now it wants to use elasticseach service. Will it call `forPlugin`?

Reply:
Services can still expose APIs to other services through their Internal contract which may be similar to the plugin-specific one.

The reason I prefer this is it moves the responsibility of constructing the exposed "public" contract to each service rather than having to constantly update the PluginsService to produce this value.